quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -237,84 +249,100 @@ public Builder(String fromProp, String toProp) {
+        public Builder<T> expectedValues(List<T> expectedValues) {
+            this.expectedValues.clear();
+            this.expectedValues.addAll(expectedValues);
             return this;
         }
 
-        public Builder expectedValues(String... expectedValues) {
+        public Builder expectedValues(T... expectedValues) {
             this.expectedValues = Arrays.asList(expectedValues);
             return this;
         }
 
-        public Builder isBuildTimeProperty(boolean isBuildTime) {
+        public Builder<T> isBuildTimeProperty(boolean isBuildTime) {
             this.isBuildTimeProperty = isBuildTime;
             return this;
         }
 
-        public Builder isMasked(boolean isMasked) {
+        public Builder<T> isMasked(boolean isMasked) {
             this.isMasked = isMasked;
             return this;
         }
 
-        public Builder category(ConfigCategory category) {
+        public Builder<T> category(OptionCategory category) {
             this.category = category;
             return this;
         }
 
-        public Builder type(Class<Boolean> type) {
+        public Builder<T> type(Class<T> type) {
             if (Boolean.class.equals(type)) {
-                expectedValues(Boolean.TRUE.toString(), Boolean.FALSE.toString());
-                paramLabel(defaultValue == null ? "true|false" : defaultValue);
-                defaultValue(defaultValue == null ? Boolean.FALSE.toString() : defaultValue);
+                expectedValues((T) Boolean.TRUE.toString(), (T) Boolean.FALSE.toString());
+                paramLabel(defaultValue == null ? "true|false" : defaultValue.toString());
+                defaultValue(defaultValue == null ? (T) Boolean.FALSE : defaultValue);
             }
+            this.type = type;
             return this;
         }
 
-        public Builder hidden(boolean hidden) {
+        public Builder<T> hidden(boolean hidden) {
             this.hidden = hidden;
             return this;
         }
 
-        public PropertyMapper build() {
-            return new PropertyMapper(from, to, defaultValue, mapper, mapFrom, isBuildTimeProperty, description, paramLabel,
+        public PropertyMapper<T> build() {
+            return new PropertyMapper<T>(type, from, to, Optional.ofNullable(defaultValue), mapper, mapFrom, isBuildTimeProperty, description, paramLabel,
                     isMasked, expectedValues, category, hidden);
         }
     }
+
+    public static <T> PropertyMapper.Builder<T> fromOption(Option<T> opt) {
+        Builder<T> builder = PropertyMapper.builder(opt.getCategory())
+                .type(opt.getType())
+                .from(opt.getKey())
+                .hidden(!opt.getSupportedRuntimes().contains(Option.Runtime.QUARKUS))
+                .description(opt.getDescription())
+                .isBuildTimeProperty(opt.isBuildTime())
+                .expectedValues(opt.getExpectedValues().stream().map(v -> v.toString()).collect(Collectors.toList()));
```

Comment:
Just out of curiosity (sorry I missed this before). Why to we need to map it to `String`? Can't we do just something like this?
```suggestion
                .expectedValues(new ArrayList(opt.getExpectedValues()));
```

Reply:
fixing it

Reply:
fixed